internal/middlewares/access_managment.go

```diff
@@ -0,0 +1,180 @@
+package middlewares
```

Comment:
Let's add unit tests for all this file at some point of time (I understand that we need to clarify idea first), adding a comment as a reminder.